service/dap/server.go

```diff
@@ -1654,10 +1654,12 @@ func (s *Server) doStepCommand(command string, threadId int, asyncSetupDone chan
 // onPauseRequest handles 'pause' request.
 // This is a mandatory request to support.
 func (s *Server) onPauseRequest(request *dap.PauseRequest) {
-	_, err := s.debugger.Command(&api.DebuggerCommand{Name: api.Halt}, nil)
-	if err != nil {
-		s.sendErrorResponse(request.Request, UnableToHalt, "Unable to halt execution", err.Error())
-		return
+	if s.debugger.IsRunning() {
```

Comment:
Should we be doing the same in StopDebugSession (like is now done in rpccommon::Stop) or it doesn't matter because there will not be another run request to halt after the end?

Reply:
Yeah it should not matter because we won't be resuming execution.